src/rustllvm/ArchiveWrapper.cpp

```diff
@@ -150,19 +158,33 @@ LLVMRustWriteArchive(char *Dst,
                      const LLVMRustArchiveMember **NewMembers,
                      bool WriteSymbtab,
                      Archive::Kind Kind) {
-  std::vector<NewArchiveIterator> Members;
+  std::vector<NewArchiveMember> Members;
 
   for (size_t i = 0; i < NumMembers; i++) {
     auto Member = NewMembers[i];
     assert(Member->name);
     if (Member->filename) {
-#if LLVM_VERSION_MINOR >= 8
-      Members.push_back(NewArchiveIterator(Member->filename));
```

Comment:
FWIW in the past I've used the precompiled 3.7/3.8 distributions that LLVM ships and I just have a few build directories when I test out these changes, I just double check the C++ compiles on each and shouldn't take too long.